<?xml version="1.0"?>
<robot name="imu_lidar_robot">

  <!-- ================= BASE LINK ================= -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.22 0.18 0.08"/>
      </geometry>
      <origin xyz="0 0 0.04" rpy="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 1 1"/>
      </material>
    </visual>
  </link>

  <!-- ================= LEFT WHEEL ================= -->
  <link name="left_wheel">
    <visual>
      <geometry>
        <cylinder radius="0.0325" length="0.02"/>
      </geometry>
      <origin xyz="0 0 0" rpy="1.5708 0 0"/>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
  </link>

<joint name="left_wheel_joint" type="continuous">
  <parent link="base_link"/>
  <child link="left_wheel"/>
  <origin xyz="0 -0.055 0.0" rpy="0 0 0"/>
  <axis xyz="0 1 0"/>
</joint>


  <!-- ================= RIGHT WHEEL ================= -->
  <link name="right_wheel">
    <visual>
      <geometry>
        <cylinder radius="0.0325" length="0.02"/>
      </geometry>
      <origin xyz="0 0 0" rpy="1.5708 0 0"/>
      <material name="black"/>
    </visual>
  </link>

<joint name="right_wheel_joint" type="continuous">
  <parent link="base_link"/>
  <child link="right_wheel"/>
  <origin xyz="0 0.055 0.0" rpy="0 0 0"/>
  <axis xyz="0 1 0"/>
</joint>

  <!-- ================= IMU ================= -->
  <link name="imu_link"/>

  <joint name="imu_joint" type="fixed">
    <parent link="base_link"/>
    <child link="imu_link"/>
    <origin xyz="0 0 0.06" rpy="0 0 0"/>
  </joint>

  <!-- ================= LIDAR ================= -->
  <link name="laser"/>

  <joint name="laser_joint" type="fixed">
    <parent link="base_link"/>
    <child link="laser"/>
    <origin xyz="0 0 0.16" rpy="0 0 0"/>
  </joint>

</robot>
